src/expr/src/relation/mod.rs

```diff
@@ -1551,6 +1594,64 @@ impl MirRelationExpr {
     }
 }
 
+// Temporary implementation for working with `LetRec`.
+impl MirRelationExpr {
+    /// True when `expr` contains a `LetRec` AST node.
+    pub fn is_recursive(self: &MirRelationExpr) -> bool {
+        let mut worklist = vec![self];
```

Comment:
This should probably be a `VecDeque`?

Reply:
I'm not sure it needs to be? The order we visit is not important, afaict. When the order is important I try to call it `stack`, but I think in this case it could be an unordered set with a `pop()` method, if that existed.